blueprints/app/files/_ts_eslint.config.mjs

```diff
@@ -0,0 +1,155 @@
+import globals from 'globals';
+import js from '@eslint/js';
+
+import ts from 'typescript-eslint';
+
+import ember from 'eslint-plugin-ember';
+import emberRecommended from 'eslint-plugin-ember/configs/recommended';
+import gjsRecommended from 'eslint-plugin-ember/configs/recommended-gjs';
+import gtsRecommended from 'eslint-plugin-ember/configs/recommended-gts';
+
+import prettier from 'eslint-plugin-prettier/recommended';
+import qunit from 'eslint-plugin-qunit';
+import n from 'eslint-plugin-n';
+
+import emberParser from 'ember-eslint-parser';
+import babelParser from '@babel/eslint-parser';
+
+const parserOptions = {
+  esm: {
+    js: {
+      ecmaFeatures: { modules: true },
+      ecmaVersion: 'latest',
+      requireConfigFile: false,
+      babelOptions: {
+        plugins: [
+          [
+            '@babel/plugin-proposal-decorators',
+            { decoratorsBeforeExport: true },
+          ],
+        ],
+      },
+    },
+    ts: {
+      projectService: true,
+      tsconfigRootDir: import.meta.dirname,
+    },
+  },
+};
+
+export default ts.config(
+  js.configs.recommended,
+  prettier,
+  {
+    files: ['**/*.js'],
+    languageOptions: {
+      parser: babelParser,
+      parserOptions: parserOptions.esm.js,
+      globals: {
+        ...globals.browser,
+      },
+    },
+    plugins: {
+      ember,
+    },
+    rules: {
+      ...emberRecommended.rules,
+    },
+  },
+  {
+    files: ['**/*.ts'],
+    plugins: { ember },
+    languageOptions: {
+      parserOptions: parserOptions.esm.ts,
+    },
+    extends: [...ts.configs.strictTypeChecked, ...emberRecommended],
+  },
+  {
+    files: ['**/*.gjs'],
+    languageOptions: {
+      parser: emberParser,
+      parserOptions: parserOptions.esm.js,
+      globals: {
+        ...globals.browser,
+      },
+    },
+    plugins: {
+      ember,
+    },
+    rules: {
+      ...emberRecommended.rules,
+      ...gjsRecommended.rules,
+    },
+  },
+  {
+    files: ['**/*.gts'],
+    plugins: { ember },
+    languageOptions: {
+      parserOptions: parserOptions.esm.ts,
+    },
+    extends: [
+      ...ts.configs.strictTypeChecked,
+      ...emberRecommended,
+      ...gtsRecommended,
+    ],
+  },
+  {
+    files: ['tests/**/*-test.{js,gjs}'],
+    plugins: {
+      qunit,
+    },
+  },
+  /**
+   * CJS node files
+   */
+  {
+    files: [
+      '**/*.cjs',
+      'config/**/*.js',
+      'testem.js',
+      'testem*.js',
+      '.prettierrc.js',
+      '.stylelintrc.js',
+      '.template-lintrc.js',
+      'ember-cli-build.js',
+    ],
+    plugins: {
+      n,
+    },
+
+    languageOptions: {
+      sourceType: 'script',
+      ecmaVersion: 'latest',
+      globals: {
+        ...globals.node,
+      },
+    },
+  },
+  /**
+   * ESM node files
+   */
+  {
+    files: ['*.mjs'],
```

Comment:
maybe 
```suggestion
    files: ['**/*.mjs'],
```
like it's done on line 107 for .cjs?

Reply:
We don't know that we want to support more depths of mjs tho?

Reply:
@NullVoxPopuli fine by me, just was confused why we would support nested .cjs but not nested .mjs

Reply:
yeah, for me, it's about wanting to get rid of mjs soonish (with type=module)

Reply:
✅